chore: add readme quick start (#247)
guardian/prosemirror-elements: docs/quick-start.md

# Adding a new element using ProseMirror Elements
## Description
The purpose of this document is to give a quick and easy guide to creating an element for a rich text editor using prosemirror-elements.

> In this example we will demonstrating this using React with Typescript and utilising the array of functions, types and design patterns that prosemirror-elements gives us for free.

Reviewer: ```suggestion
We'll do this using React and Typescript, with the functions, types and React components that prosemirror-elements gives us for free.
```

### What is an Element?
Modelling non-text content in Prosemirror can be tricky. Prosemirror-elements provides a new kind of entity, an Element, that makes this task easier. Elements contain user-defined Fields that can model many different kinds of content, including rich text fields and arbitrary data.

An Element and its Fields are represented as first class citizens of the Prosemirror schema – for example, nested rich text fields play nicely with collaborative editing.

Prosemirror-elements provides a way of expressing an Element and its Fields in the UI framework of your choice, and we provide React bindings as a default.
What is a Field?
Each Element is made up of Fields, which represent a data type – for example, text, rich text, or custom data types. 

For example, a Pullquote Element might contain the following fields:

A richText field for styled pullquote content.
A text field for the attribution.
A custom field, role, containing type `string` to indicate the pullquote’s presentation.
A custom field, isMandatory, containing type `boolean` to indicate whether the pullquote should always be displayed.

> # Example: creating a pullquote Element in React

Reviewer: ```suggestion
# Example: creating a pullquote Element in React
```

By convention, we should only have one h1 (`#`) on the page, and everything else should be h2 (`##`) or smaller.

         label="Role"
         errors={errors.role}
         display="inline"
       />
     </div>
   );
 }
);
```
And that’s it! Note that we don’t need to make use of all of these Fields in our component – we’re at liberty to choose whichever presentation we like.

Our pullquote doesn’t really need it, but in other contexts we might want to work with the current values of our Fields. These are exposed in our `fieldValues` prop, so to reveal the pullquote content as we typed, we could write:
```js
const pullquoteElement = createReactElementSpec(
 pullquoteFields,
 ({ fields, fieldValues, errors }) => {
   return (
     <div>
       <FieldWrapper
         field={fields.content}
         errors={errors.content}
         headingLabel="Pullquote content"
       />
       <blockquote>
         {fieldValues.content}
       </blockquote>
     </div>
   );
 }
);
```

## Where to keep our element
By convention, we’ve defined and kept our elements in `src/elements`. Let’s add our pullquote there, in `src/elements/example-pullquote/ExamplePullquote.tsx`, and export the element we’ve just created:
```js
export const pullquoteElement = createReactElementSpec(
  // …
```

> # Creating the Element Form

Reviewer: Is this section still necessary? We've introduced these concepts further up.

Author: Of course we have done this in 1 file
